tools/docker_activate_rses.sh

```diff
@@ -22,9 +22,19 @@ rucio-admin rse add XRD2
 rucio-admin rse add XRD3
 
 # Add the protocol definitions for the storage servers
-rucio-admin rse add-protocol --hostname xrd1 --scheme root --prefix //rucio --port 1094 --impl rucio.rse.protocols.gfal.Default --domain-json '{"wan": {"read": 1, "write": 1, "delete": 1, "third_party_copy": 1}, "lan": {"read": 1, "write": 1, "delete": 1}}' XRD1
-rucio-admin rse add-protocol --hostname xrd2 --scheme root --prefix //rucio --port 1095 --impl rucio.rse.protocols.gfal.Default --domain-json '{"wan": {"read": 1, "write": 1, "delete": 1, "third_party_copy": 1}, "lan": {"read": 1, "write": 1, "delete": 1}}' XRD2
-rucio-admin rse add-protocol --hostname xrd3 --scheme root --prefix //rucio --port 1096 --impl rucio.rse.protocols.gfal.Default --domain-json '{"wan": {"read": 1, "write": 1, "delete": 1, "third_party_copy": 1}, "lan": {"read": 1, "write": 1, "delete": 1}}' XRD3
+rucio-admin rse add-protocol --hostname xrd1 --scheme root --prefix //rucio --port 1094 --impl rucio.rse.protocols.xrootd.Default --domain-json '{"wan": {"read": 1, "write": 1, "delete": 1, "third_party_copy": 1}, "lan": {"read": 1, "write": 1, "delete": 1}}' XRD1
+rucio-admin rse add-protocol --hostname xrd2 --scheme root --prefix //rucio --port 1095 --impl rucio.rse.protocols.xrootd.Default --domain-json '{"wan": {"read": 1, "write": 1, "delete": 1, "third_party_copy": 1}, "lan": {"read": 1, "write": 1, "delete": 1}}' XRD2
+rucio-admin rse add-protocol --hostname xrd3 --scheme root --prefix //rucio --port 1096 --impl rucio.rse.protocols.xrootd.Default --domain-json '{"wan": {"read": 1, "write": 1, "delete": 1, "third_party_copy": 1}, "lan": {"read": 1, "write": 1, "delete": 1}}' XRD3
+
+# Set test_container_xrd attribute for xrd containers
+rucio-admin rse set-attribute --rse XRD1 --key test_container_xrd --value True
+rucio-admin rse set-attribute --rse XRD2 --key test_container_xrd --value True
+rucio-admin rse set-attribute --rse XRD3 --key test_container_xrd --value True
+
+# Workaround, xrootd.py#connect returns with Auth Failed due to execution of the command in subprocess
+XrdSecPROTOCOL=gsi XRD_REQUESTTIMEOUT=10 xrdfs xrd1:1094 query config xrd1:1094
```

Comment:
you don't do anything with the return values here?

Reply:
@mlassnig 
I can add:
```
XrdSecPROTOCOL=gsi XRD_REQUESTTIMEOUT=10 xrdfs xrd1:1094 query config xrd1:1094 && echo "xrd1 ok" || echo "RSEAccessDenied: xrd1"
```

I've explained a bit above on why I've added this workaround here. I'm copying that text below:
 
## Issue 2: xrootd connect fails 
### Description
The execution of the command below:
https://github.com/rucio/rucio/blob/master/lib/rucio/rse/protocols/xrootd.py#L199 

via python does not work.  I dug a little deeper by running
```
cmd = '*XrdSecDEBUG=2* XrdSecPROTOCOL=gsi XRD_REQUESTTIMEOUT=10 xrdfs xrd1:1094 query config xrd1:1094'
execute(cmd)
```
The output was as follows:
```
>>> subprocess.run(cmd, shell=True, stdout=subprocess.PIPE)
sec_Client: protocol request for host xrd1 token='&P=gsi,v:10400,c:ssl,ca:5fca1cb1.0|b96dc756.0'
sec_PM: Loaded gsi protocol object from libXrdSecgsi.so
201203 15:54:23 175 secgsi_Init: option CACheck: verifyss
201203 15:54:23 175 secgsi_Init: using CA dir(s): /etc/grid-security/certificates/
201203 15:54:23 175 secgsi_Init: option CRLCheck: 1 ('use-if-available'; download? no)
201203 15:54:23 175 secgsi_Init: using CRL dir(s): /etc/grid-security/certificates/
201203 15:54:23 175 secgsi_Init: CRL information refreshed every 86400 secs
201203 15:54:23 175 secgsi_Init: trust DNS option: 1
201203 15:54:23 175 secgsi_Init: using certificate file:         /opt/rucio/etc/usercert.pem
201203 15:54:23 175 secgsi_Init: using private key file:         /opt/rucio/etc/userkey.pem
201203 15:54:23 175 secgsi_Init: proxy: file:                    /tmp/x509up_u0
201203 15:54:23 175 secgsi_Init: proxy: validity:                12:00
201203 15:54:23 175 secgsi_Init: proxy: depth of signature path: 0
201203 15:54:23 175 secgsi_Init: proxy: bits in key:             512
201203 15:54:23 175 secgsi_Init: server cert: allowed names:
Secgsi -------------------------------------------------------------------
Secgsi Mode: client
Secgsi Debug: 2
Secgsi CA dir: /etc/grid-security/certificates/
Secgsi CA verification level: verifyss
Secgsi CRL dir: /etc/grid-security/certificates/
Secgsi CRL extension: .r0
Secgsi CRL check level: try
Secgsi CRL refresh time: 86400
Secgsi Certificate: /opt/rucio/etc/usercert.pem
Secgsi Key: /opt/rucio/etc/userkey.pem
Secgsi Proxy file: /tmp/x509up_u0
Secgsi Proxy validity: 12:00
Secgsi Proxy dep length: 0
Secgsi Proxy bits: 512
Secgsi Proxy sign option: 1
Secgsi Proxy delegation option: 0
Secgsi Allowed server names: [*/]<target host name>[/*]
Secgsi Crypto modules: ssl
Secgsi Ciphers: aes-128-cbc:bf-cbc:des-ede3-cbc
Secgsi MDigests: sha1:md5
Secgsi Trusting DNS for hostname checking
Secgsi -------------------------------------------------------------------
sec_PM: Using gsi protocol, args='v:10400,c:ssl,ca:5fca1cb1.0|b96dc756.0'
201203 15:54:23 175 secgsi_XrdSecProtocolgsi: constructing: 0x7fd0300049f0
201203 15:54:23 175 secgsi_XrdSecProtocolgsi: constructing: host: xrd1
201203 15:54:23 175 secgsi_XrdSecProtocolgsi: p: gsi, plen: 4
201203 15:54:23 175 secgsi_XrdSecProtocolgsi: mode: client
201203 15:54:23 175 secgsi_XrdSecProtocolgsi: object created: v..
201203 15:54:23 175 secgsi_ParseCrypto: parsing list: ssl
201203 15:54:23 175 secgsi_ParseCrypto: found module: ssl
201203 15:54:23 175 crypto_Factory::GetCryptoFactory: shared library 'libXrdCryptossl.so' loaded
201203 15:54:23 175 sut_Rndm::GetBuffer: enter: len: 32
201203 15:54:23 175 sut_Rndm::Init: taking seed from /dev/urandom
201203 15:54:23 175 cryptossl_sslCipher::XrdCryptosslCipher: generate DH full key
201203 15:54:23 175 secgsi_ParseCAlist: parsing list: 5fca1cb1.0|b96dc756.0
201203 15:54:23 175 secgsi_GetCA: Querying cache for tag: 5fca1cb1.0:1 (timestamp:1607010863, refresh fq:86400)
201203 15:54:23 175 secgsi_GetCA: trying to load CA certificate from /etc/grid-security/certificates/5fca1cb1.0
201203 15:54:23 175 cryptossl_X509::CertType: certificate has 3 extensions
201203 15:54:23 175 cryptossl_X509::CertType: CA certificate
201203 15:54:23 175 cryptossl_X509ParseFile: certificate for '/CN=Rucio Development CA'added to the chain - ord: 1
201203 15:54:23 175 cryptossl_X509ParseFile: no RSA private key found in file /etc/grid-security/certificates/5fca1cb1.0
201203 15:54:23 175 cryptossl_ASN1toUTC:  UTC: 1565374203  isdst: 0
201203 15:54:23 175 cryptossl_ASN1toUTC:  UTC: 2429287803  isdst: 0
201203 15:54:23 175 secgsi_LoadCRL: target file: /etc/grid-security/certificates/5fca1cb1.r0
201203 15:54:23 175 cryptossl_X509Crl::Init: file /etc/grid-security/certificates/5fca1cb1.r0 does not exist - do nothing
201203 15:54:23 175 cryptossl_X509Crl::XrdCryptosslX509Crl_file: could not initialize the CRL from /etc/grid-security/certificates/5fca1cb1.r0
201203 15:54:23 175 secgsi_GetCA: CRL is missing or expired: ignoring (CRLCheck: 1)
201203 15:54:23 175 cryptossl_X509ParseFile: unable to open file (errno: 2)
201203 15:54:23 175 secgsi_QueryProxy: proxy files must have at least two certificates (found: 0)
201203 15:54:23 175 secgsi_InitProxy: Not a tty: cannot prompt for proxies - do nothing
201203 15:54:23 175 secgsi_QueryProxy: problems initializing proxy via external shell
201203 15:54:23 175 secgsi_getCredentials: error getting user proxies CF: 0x7fd02ffffbc0
201203 15:54:23 175 secgsi_ErrF: Secgsi: ErrParseBuffer: error getting user proxies: kXGS_init
sec_Client: protocol request for host xrd1 token=' '
[FATAL] Auth failed
```

The line to be concerned about is:
```
201203 15:54:23 175 secgsi_QueryProxy: problems initializing proxy via external shell
```

Which makes sense since we are executing this command in a subshell:
https://github.com/rucio/rucio/blob/28bf965377a1271d25caf1424745dd8b99eef12f/lib/rucio/common/utils.py#L463

As a result, the tests get RseAccessDenied error

### Proposed Solution
Currently, I execute the command here (not in an external shell):
https://github.com/rucio/rucio/pull/4165/files#diff-0e09f3b5aa912ee500a40d52d7cd4e628cbc14c491d78fdeba858902fbf3f1a7R35-R37
which fixes the issue.